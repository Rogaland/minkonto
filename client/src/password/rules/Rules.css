.code {
    background: lightgray;
    border-radius: 2.5px;
    padding: 1px 5px;

    width: auto;
    display: inline;
    font-size: 95%;
    font-family: monospace;

}

.passwordRules {
    line-height: 135%;
    padding: 15px;
}