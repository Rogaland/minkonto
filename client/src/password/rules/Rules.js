import React, {Component} from 'react';

import './Rules.css';
import Typography from 'material-ui/Typography';
import Paper from 'material-ui/Paper';

class Rules extends Component {


    render() {

        return (
            <Paper className="passwordRules" elevation={20}>
                <Typography type="title">
                    Passord regler
                </Typography>
                <ul>
                    <li>Passordet må være&nbsp;
                        <div className="code">8-16 tegn</div>
                    </li>

                    <li>Passordet må inneholde karakterer fra minst 3 av kategoriene under:</li>
                    <ul>
                        <li>Store bokstaver&nbsp;
                            <p className="code">A-Z</p>
                        </li>
                        <li>Små bokstaver&nbsp;
                            <p className="code">a-z</p>
                        </li>
                        <li>Tall&nbsp;
                            <p className="code">0-9</p>
                        </li>
                        <li>Spesialtegn &nbsp;
                            <div
                                className="code">{`~ ! @ # $ % ^ & * _ - + = \` | \\ ( ) { } [ ] : ; " ' < > , . ? /`}</div>
                        </li>
                    </ul>
                    <li>Kan ikke inneholde:</li>
                    <ul>
                        <li>
                            <p className="code">brukernavn</p>
                        </li>
                        <li>
                            <p className="code">fornavn</p>
                        </li>
                        <li>
                            <p className="code">etternavn</p>
                        </li>
                    </ul>

                    <li>Tillegg:</li>
                    <ul>
                        <li>De&nbsp;
                            <p className="code">12 siste passord</p>
                            lagres i historien og kan ikke brukes.
                        </li>
                        <li>Passordet må endres hver&nbsp;
                            <p className="code">90</p>
                            &nbsp;dag.
                        </li>
                        <li>Passordet kan endres etter&nbsp;
                            <p className="code">2</p>
                            &nbsp;dager.
                        </li>
                    </ul>
                </ul>

            </Paper>
        );
    }
}

export default Rules;
