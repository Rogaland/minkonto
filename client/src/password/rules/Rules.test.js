import React from 'react';
import renderer from 'react-test-renderer';

import Rules from './Rules';

describe('Rules', () => {

  it('Renders Rules component', () => {
    const tree = renderer.create(<Rules />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});