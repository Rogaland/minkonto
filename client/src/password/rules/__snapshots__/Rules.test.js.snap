// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rules Renders Rules component 1`] = `
<div
  className="MuiPaper-root-1 MuiPaper-shadow20-23 MuiPaper-rounded-2 passwordRules"
>
  <h2
    className="MuiTypography-root-28 MuiTypography-title-34"
  >
    Passord regler
  </h2>
  <ul>
    <li>
      Passordet må være 
      <div
        className="code"
      >
        8-16 tegn
      </div>
    </li>
    <li>
      Passordet må inneholde karakterer fra minst 3 av kategoriene under:
    </li>
    <ul>
      <li>
        Store bokstaver 
        <p
          className="code"
        >
          A-Z
        </p>
      </li>
      <li>
        Små bokstaver 
        <p
          className="code"
        >
          a-z
        </p>
      </li>
      <li>
        Tall 
        <p
          className="code"
        >
          0-9
        </p>
      </li>
      <li>
        Spesialtegn  
        <div
          className="code"
        >
          ~ ! @ # $ % ^ & * _ - + = \` | \\ ( ) { } [ ] : ; " ' &lt; &gt; , . ? /
        </div>
      </li>
    </ul>
    <li>
      Kan ikke inneholde:
    </li>
    <ul>
      <li>
        <p
          className="code"
        >
          brukernavn
        </p>
      </li>
      <li>
        <p
          className="code"
        >
          fornavn
        </p>
      </li>
      <li>
        <p
          className="code"
        >
          etternavn
        </p>
      </li>
    </ul>
    <li>
      Tillegg:
    </li>
    <ul>
      <li>
        De 
        <p
          className="code"
        >
          12 siste passord
        </p>
        lagres i historien og kan ikke brukes.
      </li>
      <li>
        Passordet må endres hver 
        <p
          className="code"
        >
          90
        </p>
         dag.
      </li>
      <li>
        Passordet kan endres etter 
        <p
          className="code"
        >
          2
        </p>
         dager.
      </li>
    </ul>
  </ul>
</div>
`;
